web/src/components/storage/ProposalSettingsSection.jsx

```diff
@@ -509,6 +510,162 @@ const EncryptionPasswordField = ({
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    onSubmitProp(spacePolicy);
+  };
+
+  return (
+    <Form id={id} onSubmit={onSubmit}>
+      <div className="split">
+        <span>{_("Policy to make enough free space")}</span>
+        <ToggleGroup isCompact>
+          <ToggleGroupItem
+            text={_("Delete Anything")}
+            buttonId="deletePolicy"
+            isSelected={spacePolicy === "delete"}
+            onClick={selectDelete}
+          />
+          <ToggleGroupItem
+            text={_("Resize Only")}
+            buttonId="resizePolicy"
+            isSelected={spacePolicy === "resize"}
+            onClick={selectResize}
+          />
+          <ToggleGroupItem
+            text={_("Keep Existing")}
+            buttonId="keepPolicy"
+            isSelected={spacePolicy === "keep"}
+            onClick={selectKeep}
+          />
+        </ToggleGroup>
+      </div>
+    </Form>
+  );
+};
+
+/**
+ * Allows to select SpacePolicy.
+ * @component
+ *
+ * @param {object} props
+ * @param {ProposalSettings} props.settings - Settings used for calculating a proposal.
+ * @param {boolean} [props.isLoading=false] - Whether to show the selector as loading.
+ * @param {onChangeFn} [props.onChange=noop] - On change callback.
+ *
+ * @callback onChangeFn
+ * @param {string} policy
+ */
+const SpacePolicyField = ({
+  settings,
+  isLoading = false,
+  onChange = noop
+}) => {
+  const [isFormOpen, setIsFormOpen] = useState(false);
+  const [spacePolicy, setSpacePolicy] = useState(settings.spacePolicy);
+
+  const openForm = () => setIsFormOpen(true);
+
+  const closeForm = () => setIsFormOpen(false);
+
+  const onSubmitForm = (policy) => {
+    onChange(policy);
+    setSpacePolicy(policy);
+    closeForm();
+  };
+
+  const SpacePolicySettingsButton = () => {
+    return (
+      <Tooltip
+        content={_("Configure the Space Policy")}
+        entryDelay={400}
+        exitDelay={50}
+        position="right"
+      >
+        <button aria-label={_("Space Policy settings")} className="plain-control" onClick={openForm}>
+          <Icon name="tune" size={24} />
+        </button>
+      </Tooltip>
+    );
+  };
+
+  if (isLoading) return <Skeleton width="25%" />;
+
+  let text = "";
+  switch (spacePolicy) {
+    case "delete":
+      text = _("Delete everything");
+      break;
+    case "resize":
+      text = _("Resize as needed");
+      break;
+    case "keep":
+      text = _("Keep existing stuff");
+      break;
+    default:
+      console.log("Unsupported value " + spacePolicy);
+  }
+  const value_text = sprintf(_("Policy to make free space: %s"), text);
+
+  return (
+    <div className="split">
+      {value_text}
+      <SpacePolicySettingsButton />
+      <Popup
+        aria-label={_("Space Policy settings")}
+        title={_("Space Policy")}
+        isOpen={isFormOpen}
+      >
+        <SpacePolicySettingsForm
+          id="spacePolicySettingsForm"
+          settings={spacePolicy}
```

Comment:
Here I find confusing that `settings` is not an object but a single setting, a scalar.
I would expect `policy={spacePolicy}` or `settings={ {spacePolicy} }` (if I'm getting the single-valued object syntax right)